src/cpu/vmsa.rs

```diff
@@ -313,3 +316,32 @@ impl Vmsa {
 fn vmsa_size_check() {
     STATIC_ASSERT!(size_of::<Vmsa>() == PAGE_SIZE as usize);
 }
+
+unsafe fn update_vmsa_efer_svme(va: VirtAddr, svme: bool) -> bool {
+    flush(va);
+    BARRIER!();
+
+    let vmsa: *mut Vmsa = va.as_mut_ptr();
+    let efer_va: u64 = va.as_u64() + (*vmsa).efer_offset();
+
+    let cur_efer: u64 = (*vmsa).efer();
+    let new_efer: u64 = match svme {
+        true => cur_efer | EFER_SVME,
+        false => cur_efer & !EFER_SVME,
+    };
+
+    let xchg_efer: u64 = cmpxchg(cur_efer, new_efer, efer_va);
+    BARRIER!();
+
+    // If the cmpxchg() succeeds, xchg_efer will have the cur_efer value,
+    // otherwise, it will have the new_efer value.
+    xchg_efer == cur_efer
+}
+
+pub fn clr_vmsa_efer_svme(va: VirtAddr) -> bool {
```

Comment:
Since these is now public, lets call it vmsa_clear_efer_svme and ditto for the set function, vmsa_set_efer_svme.

Reply:
done